ui/src/warp_runner/mod.rs

```diff
@@ -110,110 +123,181 @@ async fn handle_login(notify: Arc<Notify>) {
     drop(warp_cmd_rx);
+
     if let Some(warp) = warp {
         manager::run(warp, notify).await;
     } else {
         log::info!("warp_runner terminated during initialization");
     }
 }
 
-fn init_tesseract() -> Tesseract {
-    log::trace!("initializing tesseract");
-    let tess_path = STATIC_ARGS.warp_path.join(".keystore");
-    match Tesseract::from_file(&tess_path) {
-        Ok(tess) => tess,
-        Err(_) => {
-            //doesnt exist so its set
-            log::trace!("creating new tesseract");
-            let tess = Tesseract::default();
-            tess.set_file(tess_path);
-            tess.set_autosave();
-            tess
+async fn wait_for_multipass(warp: &mut manager::Warp, notify: Arc<Notify>) -> Result<(), Error> {
+    let multipass_init_done = async {
+        loop {
+            match warp.multipass.get_own_identity().await {
+                Ok(_) => return Ok(()),
+                Err(e) => match e {
+                    Error::MultiPassExtensionUnavailable => {
+                        tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+                        continue;
+                    }
+                    _ => {
+                        log::error!("multipass.get_own_identity failed: {}", e);
+                        return Err(e);
+                    }
+                },
+            }
         }
+    };
+
+    tokio::select! {
+        r = multipass_init_done => r,
+        _ = notify.notified() => {
+            log::error!("notified interrupted multipass initialization");
+            Err(Error::Other)
+        },
     }
 }
 
-// tesseract needs to be initialized before warp is initialized. this function does just that
-async fn login(
-    passphrase: &str,
-    tesseract: Tesseract,
-    new_account: bool,
-) -> Result<manager::Warp, warp::error::Error> {
-    log::debug!("login");
-    tesseract.unlock(passphrase.as_bytes())?;
-    while !tesseract.is_unlock() {
-        tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-    }
+// don't set file or autosave until tesseract is unlocked
+// assumes that all anyone needs from tesseract is "keypair"
+// otherwise, Tesseract::to_file probably needs to call file.sync_all()
+async fn init_tesseract(overwrite_old_account: bool) -> Result<Tesseract, Error> {
+    log::trace!("initializing tesseract");
+
+    let configure_tesseract = |tesseract: Tesseract| {
+        // prevent other things from corrupting the real tesseract file.
+        tesseract.set_file(STATIC_ARGS.warp_path.join("fake_tesseract.json"));
+        tesseract.set_autosave();
+        tesseract
+    };
+
+    // this code path addresses cross-platform issues involving account recreation.
+    // the tesseract file was being overwritten incorrectly.
+    // to fix this, manually delete the file and re-create it.
+    if overwrite_old_account {
+        // delete old account data
+        if let Err(e) = std::fs::remove_dir_all(&STATIC_ARGS.uplink_path) {
```

Comment:
Not required since these are short lived, but may want to use `tokio::fs` for these since its inside of an async function